File: LtpCsharp/Postagger.cs
using LtpCsharp.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LtpCsharp
{
    public class Postagger 
    {
        IPostagger native = null;

        public Postagger(string path)
        {
            native = LtpFactory.GetLtpFactory().CreatePostagger(path);
        }
        public Postagger(string path, string lexicon_file)
        {
            native = LtpFactory.GetLtpFactory().CreatePostagger(path, lexicon_file);
        }

        public int Postag(List<string> words, out List<string> tags)
        {
            return native.Postag(words, out tags);
        }
    }
}

File: ltpcsharp.native/Postagger.cpp
#include "stdafx.h"
#include "ltpcsharp.h"
#include <string>
#include <memory>
#include <locale>
#include <codecvt>
#include <msclr/marshal.h>
#include <msclr/marshal_cppstd.h>
#include "../ltpinclude/postag_dll.h"

using namespace System;
using namespace System::Collections::Generic;
using namespace System::Runtime::InteropServices;

namespace ltpcsharp_native {
	
	public ref class Postagger
	{
		void * wrapper = nullptr;
	public:
		Postagger(String^path, String^ lexicon_file)
		{
			std::string converted_path = msclr::interop::marshal_as< std::string >(path);
			std::string converted_lexicon_file = msclr::interop::marshal_as< std::string >(lexicon_file);
			wrapper = postagger_create_postagger(converted_path.c_str(), converted_lexicon_file.c_str());
			if (wrapper == nullptr)
			{
				throw gcnew ArgumentException("参数错误,初始化失败");
			}
		}
		Postagger(String^path)
		{
			std::string converted_path = msclr::interop::marshal_as< std::string >(path);
			wrapper = postagger_create_postagger(converted_path.c_str());
			if (wrapper == nullptr)
			{
				throw gcnew ArgumentException("参数错误,初始化失败");
			}
		}
		~Postagger()
		{
			postagger_release_postagger(wrapper);
			wrapper = nullptr;
		}
		!Postagger()
		{
			postagger_release_postagger(wrapper);
			wrapper = nullptr;
		}

		int Postag([In] List<String^>^ words,
			[Out] List<String^>^% tags)
		{
			auto cppwords = Help::CSharpToCpp(words);
			std::vector<std::string> cpptags;
			int doret = postagger_postag(wrapper, cppwords, cpptags);
			tags = Help::CppToCSharp(cpptags);
			return doret;
		}
	};
}